src/System.Linq/src/System/Linq/OrderedEnumerable.cs

```diff
@@ -71,6 +71,57 @@ public List<TElement> ToList()
             return list;
         }
 
+        public HashSet<TElement> ToHashSet(IEqualityComparer<TElement> comparer)
+        {
+            Buffer<TElement> buffer = new Buffer<TElement>(_source);
```

Comment:
Is it really necessary to retain the ordering within the hash set? While it is reflected on immediate enumeration, that's of little value, not specified, and hard to predict after subsequent operations.

```C#
public HashSet<TElement> ToHashSet(IEqualityComparer<TElement> comparer) => new HashSet<TElement>(_source, comparer);